src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// Shared code to run the "sources" command from the command line projects
+    /// </summary>
+    public static class SourcesRunner
+    {
+        public static void Run(SourcesArgs args)
+        {
+            switch (args.Action)
+            {
+                case SourcesAction.List:
+                    switch (args.Format)
+                    {
+                        case CommandLine.SourcesListFormat.Short:
+                            PrintRegisteredSourcesShort(args);
+                            break;
+                        case CommandLine.SourcesListFormat.Detailed:
+                        default:
+                            PrintRegisteredSourcesDetailed(args);
+                            break;
+                    }
+                    break;
+                case SourcesAction.Add:
+                    AddNewSource(args);
+                    break;
+                case SourcesAction.Remove:
+                    RemoveSource(args);
+                    break;
+                case SourcesAction.Enable:
+                    EnableOrDisableSource(args, enabled: true);
+                    break;
+                case SourcesAction.Disable:
+                    EnableOrDisableSource(args, enabled: false);
+                    break;
+                case SourcesAction.Update:
+                    UpdatePackageSource(args);
+                    break;
+            }
+        }
+
+        private static void EnableOrDisableSource(SourcesArgs args, bool enabled)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
+
+            var packageSource = args.SourceProvider.GetPackageSourceByName(args.Name);
+            if (packageSource == null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNoMatchingSourcesFound, args.Name));
+            }
+
+            if (enabled && !packageSource.IsEnabled)
+            {
+                args.SourceProvider.EnablePackageSource(args.Name);
+            }
+            else if (!enabled && packageSource.IsEnabled)
+            {
+                args.SourceProvider.DisablePackageSource(args.Name);
+            }
+
+            if (args.Verbosity != CommandLine.Verbosity.Quiet)
+            {
+                if (enabled)
+                {
+                    Console.WriteLine(string.Format(CultureInfo.CurrentCulture,
```

Comment:
I wouldn't call console write line directly, I'd pass a delegate.

Reply:
Can do that similarly to Log.Error, etc...

Reply:
passing in IsQuiet and SourceArgs has an implementation of a log delegate that uses IsQuiet